lib/Github/Client.php

```diff
@@ -336,17 +305,13 @@ public function api($name)
      */
     public function authenticate($tokenOrLogin, $password = null, $authMethod = null)
     {
-        if (null === $password && null === $authMethod) {
-            throw new InvalidArgumentException('You need to specify authentication method!');
-        }
-
-        if (null === $authMethod && in_array($password, [self::AUTH_URL_TOKEN, self::AUTH_URL_CLIENT_ID, self::AUTH_HTTP_PASSWORD, self::AUTH_HTTP_TOKEN, self::AUTH_JWT], true)) {
```

Comment:
Shouldn't we keep the `in_array` check to verifiy users provide a supported method?

Reply:
Yeh, something's got lost in the refactoring here. I'll take a look. :)